core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.kt

```diff
@@ -0,0 +1,284 @@
+import org.kiwix.kiwixmobile.core.utils.EXTRA_CHOSE_X_FILE
+import org.kiwix.kiwixmobile.core.utils.EXTRA_CHOSE_X_TITLE
+import org.kiwix.kiwixmobile.core.utils.EXTRA_CHOSE_X_URL
+import org.kiwix.kiwixmobile.core.utils.KiwixDialog
+import java.util.ArrayList
+import javax.inject.Inject
+
+class BookmarksActivity : BaseActivity(),
+  BookmarksContract.View, BookmarksAdapter.OnItemClickListener {
+  private val bookmarksList: MutableList<BookmarkItem> = ArrayList()
+  private val allBookmarks: MutableList<BookmarkItem> = ArrayList()
+  private val deleteList: MutableList<BookmarkItem> = ArrayList()
+
+  @JvmField @BindView(R2.id.toolbar)
+  var toolbar: Toolbar? = null
+
+  @JvmField @BindView(R2.id.recycler_view)
+  var recyclerView: RecyclerView? = null
+
+  @JvmField @BindView(R2.id.no_bookmarks)
+  var noBookmarks: TextView? = null
+
+  @JvmField @BindView(R2.id.bookmarks_switch)
+  var bookmarksSwitch: Switch? = null
+
+  @Inject
+  var presenter: BookmarksContract.Presenter? = null
+
+  @Inject
+  var zimReaderContainer: ZimReaderContainer? = null
+
+  @Inject
+  var dialogShower: DialogShower? = null
+  private var refreshAdapter = true
+  private var bookmarksAdapter: BookmarksAdapter? = null
+  private var actionMode: ActionMode? = null
+  private val actionModeCallback: ActionMode.Callback =
+    object : ActionMode.Callback {
+      override fun onCreateActionMode(
+        mode: ActionMode,
+        menu: Menu
+      ): Boolean {
+        mode.menuInflater.inflate(R.menu.menu_context_delete, menu)
+        bookmarksSwitch!!.isEnabled = false
+        return true
+      }
+
+      override fun onPrepareActionMode(
+        mode: ActionMode,
+        menu: Menu
+      ): Boolean = false
+
+      override fun onActionItemClicked(
+        mode: ActionMode,
+        item: MenuItem
+      ): Boolean {
+        refreshAdapter = false
+        if (item.itemId == R.id.menu_context_delete) {
+          dialogShower!!.show(KiwixDialog.DeleteBookmarks, {
+            allBookmarks.removeAll(deleteList)
+            for (bookmark in deleteList) {
+              val position = bookmarksList.indexOf(bookmark)
+              bookmarksList.remove(bookmark)
+              bookmarksAdapter!!.notifyItemRemoved(position)
+              bookmarksAdapter!!.notifyItemRangeChanged(position, bookmarksAdapter!!.itemCount)
+            }
+            presenter!!.deleteBookmarks(ArrayList(deleteList))
+            mode.finish()
+          })
+          return true
+        }
+        return false
+      }
+
+      override fun onDestroyActionMode(mode: ActionMode) {
+        if (deleteList.size != 0) {
+          deleteList.clear()
+        }
+        actionMode = null
+        if (refreshAdapter) {
+          bookmarksAdapter!!.notifyDataSetChanged()
+        }
+        bookmarksSwitch!!.isEnabled = true
+      }
+    }
+
+  override fun injection(coreComponent: CoreComponent) {
+    coreComponent.activityComponentBuilder().activity(this).build().inject(this)
```

Comment:
there should be an extension function, `coreActivityComponent` that will simplify this call